lib/item/download_items_helper.rb

```diff
@@ -123,26 +145,86 @@ def createAndRetrieveWarcPath(itemHandles, url, &block)
 
     private
 
+    def get_aspera_transfer_spec_for_documents_and_metadata(item_list, transfer_dir)
+      item_handles = item_list.get_item_handles
+      return nil if item_handles.empty?
+
+      result = verify_items_permissions_and_extract_metadata(item_handles)
+
+      # get item files and create metadata files and log files
+      item_files = get_items_files(result)
+      metadata_files = create_items_metadata_files(result, transfer_dir)
+      log_file = create_items_log_file(result, transfer_dir)
+
+      request_aspera_transfer_spec(item_list.name, [item_files, metadata_files, log_file].flatten)
+    end
+
+    def get_items_files(result)
+      filenames = get_filenames_from_item_results(result)
+      filenames.map do |key, value|
+        dir = value[:handle]
+        files = value[:files]
+        files.map { |file| { dir: dir, file: file } }
+      end.flatten
+    end
+
+    def create_items_metadata_files(result, transfer_dir)
+      items_metadata = result[:metadata]
+      items_metadata.map do |key, value|
+        metadata = value[:metadata]
+        handle = metadata['metadata']['handle'].gsub(":", "_")
+        file = File.join(transfer_dir, "#{handle}/#{handle}-metadata.json")
+
+        # write metadata to file
+        FileUtils.mkdir_p File.dirname(file)
+        File.open(file , 'w+') do |f|
+          f.write(metadata.to_json)
+        end
+
+        { dir: handle, file: file }
+      end
+    end
+
+    def create_items_log_file(result, transfer_dir)
+      file = File.join(transfer_dir, 'log.json')
+      File.open(file, 'w+') do |f|
+        f.write(generate_json_log(result[:valids], result[:invalids]))
+      end
+      { dir: '.', file: file }
+    end
+
+    def request_aspera_transfer_spec(items_dir, requested_files)
+      source_paths = requested_files.map {|file| { source: file[:file], destination: "/#{items_dir}/#{file[:dir]}/#{File.basename(file[:file])}" }}
+      download_request = {
+        transfer_requests: [
+          transfer_request: {
+            source_root: '',
+            paths: source_paths
+          }
+        ]
+      }
+      node_api = NodeAPI.new(aspera_nodeapi_config)
```

Comment:
Why not have the config as a default parameter in the NodeAPI constructor? It seems unlikely that any other parameters would be passed in except perhaps for testing. Also, node_api is constructed for each call, perhaps this would be better off as an instance variable (then again, it doesn't look like it's used anywhere else).